metron-interface/metron-rest/src/main/java/org/apache/metron/rest/service/GrokService.java

```diff
@@ -0,0 +1,144 @@
+        } catch (StringIndexOutOfBoundsException e) {
+            results.put("error", "A pattern label must be included (ex. PATTERN_LABEL ${PATTERN:field} ...)");
+        } catch (Exception e) {
+            results.put("error", e.getMessage());
+        }
+        grokValidation.setResults(results);
+        return grokValidation;
+    }
+
+    public boolean isGrokConfig(SensorParserConfig sensorParserConfig) {
+        return GROK_CLASS_NAME.equals(sensorParserConfig.getParserClassName());
+    }
+
+    public void addGrokStatementToConfig(SensorParserConfig sensorParserConfig) throws IOException {
+        String grokStatement = "";
+        String grokPath = (String) sensorParserConfig.getParserConfig().get(GROK_PATH_KEY);
+        if (grokPath != null) {
+            try {
+                String fullGrokStatement = getGrokStatement(grokPath);
+                String patternLabel = (String) sensorParserConfig.getParserConfig().get(GROK_PATTERN_LABEL_KEY);
+                grokStatement = fullGrokStatement.replaceFirst(patternLabel + " ", "");
+            } catch(FileNotFoundException e) {}
+        }
+        sensorParserConfig.getParserConfig().put(GROK_STATEMENT_KEY, grokStatement);
+    }
+
+    public void addGrokPathToConfig(SensorParserConfig sensorParserConfig) {
+        if (sensorParserConfig.getParserConfig().get(GROK_PATH_KEY) == null) {
+            String grokStatement = (String) sensorParserConfig.getParserConfig().get(GROK_STATEMENT_KEY);
+            if (grokStatement != null) {
+              sensorParserConfig.getParserConfig().put(GROK_PATH_KEY,
+                      new Path(environment.getProperty(GROK_PATH_SPRING_PROPERTY), sensorParserConfig.getSensorTopic()).toString());
+            }
+        }
+    }
+
+    public String getGrokStatement(String path) throws IOException {
+        return new String(hdfsService.read(new Path(path)));
+    }
+
+    public void saveGrokStatement(SensorParserConfig sensorParserConfig) throws IOException {
+        saveGrokStatement(sensorParserConfig, false);
+    }
+
+    public void saveTemporaryGrokStatement(SensorParserConfig sensorParserConfig) throws IOException {
+        saveGrokStatement(sensorParserConfig, true);
+    }
+
+    private void saveGrokStatement(SensorParserConfig sensorParserConfig, boolean isTemporary) throws IOException {
+        String patternLabel = (String) sensorParserConfig.getParserConfig().get(GROK_PATTERN_LABEL_KEY);
+        String grokPath = (String) sensorParserConfig.getParserConfig().get(GROK_PATH_KEY);
+        String grokStatement = (String) sensorParserConfig.getParserConfig().get(GROK_STATEMENT_KEY);
+        String fullGrokStatement = patternLabel + " " + grokStatement;
+        if (grokStatement != null) {
+            if (!isTemporary) {
+                hdfsService.write(new Path(grokPath), fullGrokStatement.getBytes());
+            } else {
+                FileWriter fileWriter = new FileWriter(new File(grokPath));
```

Comment:
What if multiple users are using this simultaneously?  Won't these temp files conflict?